hpcpy/client/pbs.py

```diff
@@ -84,37 +111,31 @@ def submit(
             Key/value environment variable pairs added to the qsub command.
         **context:
             Additional key/value pairs to be added to command/jobscript interpolation
-        """
 
-        directives = directives if isinstance(directives, list) else list()
+        Returns
+        -------
+        Job : hpcpy.job.Job
+            Job object.
+        """
+        # Initialise the directives to an empty list
+        directives = directives if isinstance(directives, list) else []
 
         # Add job depends
         if depends_on:
-            depends_on = hu.ensure_list(depends_on)
-            directives.append("-W depend=afterok:" + ":".join(depends_on))
+            directives = self._interpolate_directive(
+                directives,
+                "depends_on",
+                depends_on_str=":".join(hu.ensure_list(depends_on)),
+            )
 
         # Add delay (specified time or delta)
         if delay:
-
-            current_time = datetime.now()
-            delay_str = None
-
-            if isinstance(delay, datetime) and delay > current_time:
-                delay_str = delay.strftime("%Y%m%d%H%M.%S")
-
-            elif isinstance(delay, timedelta) and (current_time + delay) > current_time:
-                delay_str = (current_time + delay).strftime("%Y%m%d%H%M.%S")
-            else:
-                raise ValueError(
-                    "Job submission delay argument either incorrect or puts the job in the past."
-                )
-
-            # Add the delay directive
-            directives.append(f"-a {delay_str}")
+            delay_directive = self._assemble_delay_directive(delay)
```

Comment:
As per delay_directive_fmt suggestion in base class
```suggestion
            delay_directive = self._assemble_delay_directive(delay, delay_directive_fmt="%Y%m%d%H%M.%S")
```